net-print/kyocera-1x2x-mfp-driver/kyocera-1x2x-mfp-driver-1.1203-r1.ebuild

```diff
@@ -0,0 +1,134 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+DESCRIPTION="Printer descriptions (PPDs) and filters for Kyocera 1x2x MFP"
+HOMEPAGE="http://www.kyoceradocumentsolutions.eu"
+SRC_URI="LinuxDrv_${PV}_FS-1x2xMFP.zip"
+
+LICENSE="kyocera-mita-ppds"
+SLOT="0"
+
+KEYWORDS="-* ~amd64"
+
+IUSE_L10N=(en ar cs de el es fr he hu it ko pl pt ro ru th tr vi zh-CN zh-TW)
+IUSE="+rastertokpsl-fix ${IUSE_L10N[@]/#/l10n_}"
+REQUIRED_USE="|| ( ${IUSE_L10N[@]/#/l10n_} )"
+
+RDEPEND="net-print/cups"
+DEPEND="app-arch/unzip"
+RESTRICT="fetch mirror"
+QA_PREBUILT="/usr/libexec/cups/filter/rastertokpsl"
+
+get_tarball_name() {
+	# Note the capitalization inconsistency. Don't "fix" that.
+	case "$1" in
+		ar) echo arabic ;;
+		cs) echo czech ;;
+		de) echo German ;;
+		el) echo greek ;;
+		en) echo English ;;
+		es) echo Spanish ;;
+		fr) echo French ;;
+		he) echo hebrew ;;
+		hu) echo hungarian ;;
+		it) echo Italian ;;
+		ko) echo Korean ;;
+		pl) echo polish ;;
+		pt) echo Portuguese ;;
+		ro) echo romanian ;;
+		ru) echo russian ;;
+		th) echo thai ;;
+		tr) echo turkish ;;
+		vi) echo vietnamese ;;
+		zh-CN) echo simplified ;;
+		zh-TW) echo traditional ;;
+		*) die ;;
+	esac
+}
+
+pkg_nofetch() {
+	einfo "Please, navigate your browser to the following URL and manually"
+	einfo "download the file named '${A}', then put it into '${DISTDIR}'."
+	einfo "http://www.kyoceradocumentsolutions.eu/index/service/dlc.false.driver.FS1025MFP._.EN.html"
+	einfo
+	einfo "Consider keeping a local copy of the file since there're chances"
+	einfo "the company is going to eventually stop hosting it for whatever"
+	einfo "reason."
+}
+
+src_unpack() {
+	unpack "${A}"
+	S="${WORKDIR}/Linux/$(usex amd64 64bit 32bit)/Global"
+
+	cd "${S}" || die
+	local l10n
+	for l10n in "${IUSE_L10N[@]}"; do
+		if use "l10n_${l10n}"; then
+			local language="$(get_tarball_name "${l10n}")"
+			unpack "./${language}.tar.gz"
+		fi
+	done
+}
+
+src_prepare() {
+	eapply_user
+
+	patch_filter_path() {
+		local orig="/usr/lib/cups/filter/rastertokpsl"
+		local repl="/usr/libexec/cups/filter/rastertokpsl"
+		if use rastertokpsl-fix; then
+			repl+="-fix.sh"
+		fi
+
+		sed -i "s|${orig}|${repl}|g" "$@" || die
+	}
+
+	patch_filter_path ./*/Kyocera*.ppd
```

Comment:
Actually, why do you need a sub-function for this? After all, you just pass all paths to sed, so you could just inline it.

Reply:
From my non-shell programming background, having a separate function for a well-defined task feels like a "right" thing even if I have to call it only once. If only Bash had the notion of closures and nested functions 😢. (The fact that this "nested" functions are actually global made me name them in camel case originally, so nothing can interfere with them by accident.)

I don't insist on having this as a function here, but I believe it's making the code more readable. The same also applies for `install_with_l10n_suffix`.

Reply:
Well, lack of local functions is exactly why I don't like it here. It's causing global env pollution for no real advantage.

Reply:
BTW, it's `kyocera-mita-ppds-8.4.ebuild` where I borrowed the "use nested function if you want"-approach. I thought it's a kind of well-established practice in the portage tree.

Reply:
I'm fine with inlining `patch_filter_path`, but getting rid of `install_with_l10n_suffix` will make the code look much worse. Can we just prefix the name with underscore (like in Python), or `aux_`, or something else to denote it's for internal use? 

Or file a patch to the Bash mailing list to get actual support for nested functions ^__^.

Reply:
The latter function is fine, it at least has a valid case for calling multiple times. To make it feel more local, you can `unset -f` it after use.